packages/react/src/hooks/actions/useAuthSignOutAction.tsx

```diff
@@ -0,0 +1,33 @@
+import { Auth, Hub } from 'aws-amplify';
+import { SignOutOpts } from '@aws-amplify/auth/lib-esm/types/Auth';
+
+import {
+  ACTION_EVENT_FINISHED,
+  ACTION_EVENT_STARTED,
+  HUB_AUTH_SIGNOUT_NAMESPACE,
+} from './constants';
+import { getErrorMessage } from '../../helpers/utils';
+
+interface UseAuthSignOutAction {
+  (opts?: SignOutOpts): () => Promise<void>;
+}
+
+export const useAuthSignOutAction: UseAuthSignOutAction =
+  (opts) => async () => {
+    try {
+      Hub.dispatch(HUB_AUTH_SIGNOUT_NAMESPACE, {
+        event: ACTION_EVENT_STARTED,
```

Comment:
Don't you want the opts also attached to this as well? We should probably indicate whether or not it was global in the callback hooks.

Reply:
Good idea, will update